core-v1/src/main/scala/data/Justified.scala

```diff
@@ -169,16 +187,30 @@ object Justified {
 
   implicit def extractValue[V]: ExtractValue[Justified[V], V] = _.value
 
-  implicit val functor: Functor[Justified] = new Functor[Justified] {
-    override def map[A, B](fa: Justified[A])(f: A => B): Justified[B] = fa match {
-      case Justified.ByConst(v) => Justified.byConst(f(v))
-      case Justified.ByConfig(v, k, d) => Justified.byConfig(f(v), k, d)
+  implicit val wrapConst: WrapConst[Justified] = new WrapConst[Justified] {
+    override def wrapConst[A](value: A): Justified[A] = Justified.byConst(value)
+  }
+
+  // TODO: Include whether a collection or not in WrapSelected interface
+  private final case object WrapSelectedJustified extends WrapSelected[Justified, Any] {
+    override def wrapSelected[I, O](
+      container: Justified[I],
+      path: DataPath,
+      element: O,
+    )(implicit
+      opA: Any,
+      opB: Any,
+    ): Justified[O] = container match {
+      case Justified.BySelection(_, prevPath, prevSource) =>
+        Justified.bySelection(element, prevPath ++ path, prevSource)
       case _ =>
-        val derived = f(fa.value)
-        Justified.byInference("map", derived, NonEmptyList.of(fa))
+        Justified.bySelection(element, path, container)
     }
```

Comment:
Do or remove TODO. I think the goal here is to know whether the container is a collection or not, and, if so, which index into that container the value is being selected.

Reply:
I think this should is now solved by the `SelectOutputType` traits. The only thing that is missing is support for `Map` selection, but I'll tackle that in a different PR.